.github/workflows/whitespace.yml

```diff
@@ -14,13 +14,12 @@ jobs:
     strategy:
       matrix:
         ghc-ver: [9.0.1]
-        stack-ver: [2.5.1]
+        cabal-ver: [3.4.0.0]
+        # stack-ver: [2.5.1]
 
     steps:
     - uses: actions/checkout@v2
-    - name: Check the commit SHA of submodule fix-whitespace
-      run: |
-        echo fw-sha=`git submodule status -- src/fix-whitespace | git submodule status -- src/fix-whitespace | cut -c 2- | cut -d " " -f 1`  >> $GITHUB_ENV
```

Comment:
Removing this step is not a good idea. The cache key is the environment variable `fw-sha` exported in this step. This is used to check if the workflow is using the latest `fix-whitespace`. If not, then it will recompile the binary.

Reply:
Thanks for the swift review, @L-TChen !!

Ah, sorry, I did not look carefully here.
Since `fix-whitespace` is no longer a submodule, we have to find another way to get a cache-key.
We could maybe use the latest published version of `fix-whitespace` as key 
```
$ cabal info fix-whitespace | grep "Versions available" | awk '{ print $NF }'
0.0.5
```
(This command is a bit brittle, but `cabal` does not seem to offer a command that gets the versions of a package from hackage, and I do not want to rely on experimental tools like `hackacke-cli`.)
What do you think?

Reply:
This doesn't work, since the cabal package list has not yet set up. The step `haskell/actions/setup@v1` is used to set up cabal, but running `cabal update` itself is too expensive just for checking the latest version. 

```
Warning: The package list for 'hackage.haskell.org' does not exist. Run 'cabal
5
update' to download it.RemoteRepo {remoteRepoName = RepoName
6
"hackage.haskell.org", remoteRepoURI = http://hackage.haskell.org/,
7
remoteRepoSecure = Just True, remoteRepoRootKeys =
8
["fe331502606802feac15e514d9b9ea83fee8b6ffef71335479a2e68d84adc6b0","1ea9ba32c526d1cc91ab5e5bd364ec5e9e8cb67179a471872f6e26f0ae773d42","2c6c3627bd6c982990239487f1abd02e08a02e6cf16edb105a8012d444d870c3","0a5c7ea47cd1b15f01f5f51a33adda7e655bc0f0b0615baa8e271f4c3351e21d","51f0161b906011b52c6613376b1ae937670da69322113a246a09f807c62f6921"],
9
remoteRepoKeyThreshold = 3, remoteRepoShouldTryHttps = True}
10
cabal: There is no package named 'fix-whitespace'.
11
You may need to run 'cabal update' to get the latest list of available
12
packages.
```

Perhaps we just make the version of `fix-whitespace` a parameter like `ghc-ver` and `cabal-ver` and update the version whenever necessary. It shouldn't be updated as regularly as `ghc` or `cabal`, so it is fine I believe.

Reply:
Ok, I now added a new matrix entry `fix-whitespace-ver` and use this as cache key.

Reply:
The workflow seems to work again.

Reply:
Now it looks good to me!